submissions/YuliiaHordiichuk/post-oop/index.js

```diff
@@ -0,0 +1,80 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/YuliiaHordiichuk/a-tiny-JS-world/tree/post-oop
+   Web app: https://yuliiahordiichuk.github.io/a-tiny-JS-world/
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+class Inhabitant {
+   constructor(name, gender, saying, species) {
+      this.name = name; 
+      this.gender = gender; 
+      this.saying = saying; 
+      this.species = species; 
+   };
+
+   getMessage() {
+      const messageText = [
+      this.saying,
+      `I am a ${this.species}.`, 
+      `My name is ${this.name}.`,
+      `Nice to meet you!`,
+      `My gender is ${this.gender}.`
+   ]; 
```

Comment:
If you rename `getMessage` into `toString` you will have a magic method.
`toString` is a method that is called implictly in string context. `print` ensures string context for its arguments.

`toString` is defined for every object in JS. It is `toString` that is responsible for output of `[Object object]` in some contexts (e.g. ``console.log(`Logging an object: ${obj}`)``). It is overloadable and we can benefit from this.

Reply:
Didn`t know about this, thanks!